### PR TITLE
[SPARK-31739][PYSPARK][DOCS][MINOR] Fix docstring syntax issues and misplaced space characters.

### PR DESCRIPTION
**Pull request by DavidToneian**
This commit is published into the public domain.

### What changes were proposed in this pull request?
Some syntax issues in docstrings have been fixed.

### Why are the changes needed?
In some places, the documentation did not render as intended, e.g. parameter documentations were not formatted as such.

### Does this PR introduce any user-facing change?
Slight improvements in documentation.

### How was this patch tested?
Manual testing and `dev/lint-python` run. No new Sphinx warnings arise due to this change.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]

**Review by srowen — APPROVED** [review]

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -2150,9 +2150,9 @@ def toDF(self, *cols):
 
     @since(3.0)
     def transform(self, func):
-        """Returns a new class:`DataFrame`. Concise syntax for chaining custom transformations.
+        """Returns a new :class:`DataFrame`. Concise syntax for chaining custom transformations.
 
-        :param func: a function that takes and returns a class:`DataFrame`.
+        :param func: a function that takes and returns a :class:`DataFrame`.
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/dataframe.py:2155]: Could you fix `classification.py` and `regression.py`, too?
```
pyspark/ml/classification.py:    To be mixed in with class:`pyspark.ml.JavaModel`
pyspark/ml/regression.py:    To be mixed in with class:`pyspark.ml.JavaModel`
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (with one minor comment). Thank you, @DavidToneian .

**Comment by DavidToneian** [comment]
@dongjoon-hyun Thanks for pointing those out, fixed in the new commit above.

**Comment by HyukjinKwon** [comment]
Merged to master.

I will pick this up when RC is cut out.

**Comment by HyukjinKwon** [comment]
Merged to branch-3.0 as well.

**Pull request closed without merging**